### PR TITLE
Adding search request timeout examples

### PR DESCRIPTION
**Pull request by sreekanth-cb**

### CODE REVIEW

**Comment by sreekanth-cb** [comment]
here is the new one @abhinavdangeti .

Tagging @amarantha-k too.

File: modules/fts/pages/fts-searching-with-the-rest-api.adoc [hunk 1]
```diff
@@ -100,6 +100,41 @@ http://localhost:18094/api/index/travel-sample-index/query \
 For additional information on secure communication with Couchbase Server, see xref:manage:manage-security/manage-connections-and-disks.adoc[Manage Connections and Disks].
 
 
+[#searching-with-request-timeouts]
+== Searching with timeouts
+
+Once the the index becomes really large, it is possible that certain search requests may take
```
Comment (amarantha-k) [inline modules/fts/pages/fts-searching-with-the-rest-api.adoc]: repeated "the the"

File: modules/fts/pages/fts-searching-with-the-rest-api.adoc [hunk 2]
```diff
@@ -100,6 +100,41 @@ http://localhost:18094/api/index/travel-sample-index/query \
...
+longer time to finish the execution. Such slow responses would affect the query latencies as well as the resource consumption on the server side.
```
Comment (amarantha-k) [inline modules/fts/pages/fts-searching-with-the-rest-api.adoc]: Minor editorial comments:
- may take a longer time to complete execution. 
- Such slow responses affect query latencies as well as resource consumption on the server side.

File: modules/fts/pages/fts-searching-with-the-rest-api.adoc [hunk 3]
```diff
@@ -100,6 +100,41 @@ http://localhost:18094/api/index/travel-sample-index/query \
...
+
+In order to handle such situations predictably, users can pass a search "timeout" value in the request.
+By specificing a timeout parameters in the seach request, users are bound to get a response within the stipulated time.
```
Comment (amarantha-k) [inline modules/fts/pages/fts-searching-with-the-rest-api.adoc]: a timeout parameter

File: modules/fts/pages/fts-searching-with-the-rest-api.adoc [hunk 4]
```diff
@@ -100,6 +100,41 @@ http://localhost:18094/api/index/travel-sample-index/query \
...
+Here the timeout value is specified in milliseconds.
```
Comment (amarantha-k) [inline modules/fts/pages/fts-searching-with-the-rest-api.adoc]: The timeout value...

File: modules/fts/pages/fts-searching-with-the-rest-api.adoc [hunk 5]
```diff
@@ -100,6 +100,41 @@ http://localhost:18094/api/index/travel-sample-index/query \
...
+
+If the search request times out before gathering all the results, then the partial results will be
+returned to the user with detailed partition level search result status.
+In case of timeouts, the response's http status code would be 412(Precondition Failed).
```
Comment (amarantha-k) [inline modules/fts/pages/fts-searching-with-the-rest-api.adoc]: HTTP

File: modules/fts/pages/fts-searching-with-the-rest-api.adoc [hunk 6]
```diff
@@ -100,6 +100,41 @@ http://localhost:18094/api/index/travel-sample-index/query \
...
+  "ctl": {
+    "timeout": 7000
+  },
+  "query": {
+    "query": "{ \"+nice +view\" }"
+  }
+}'
+----
+
+Please note that the default value of timeout is `10` seconds and users may override this
```
Comment (amarantha-k) [inline modules/fts/pages/fts-searching-with-the-rest-api.adoc]: Note that... and you may change this value to suit your use case.

(General recommendations:
- avoid using "please" 
- write in second person i.e., refer to users as "you" )

**Comment by sreekanth-cb** [comment]
done @amarantha-k , thanks!

**Review by amarantha-k — APPROVED** [review]
thanks for incorporating the feedback. lgtm

**Pull request merged**